src/sys.rs

```diff
@@ -28,22 +28,22 @@ impl ::std::default::Default for __va_list_tag {
 
 pub const JNI_FALSE: ::std::os::raw::c_uchar = 0;
 pub const JNI_TRUE: ::std::os::raw::c_uchar = 1;
-pub const JNI_OK: ::std::os::raw::c_uchar = 0;
-pub const JNI_ERR: ::std::os::raw::c_char = -1;
-pub const JNI_EDETACHED: ::std::os::raw::c_char = -2;
-pub const JNI_EVERSION: ::std::os::raw::c_char = -3;
-pub const JNI_ENOMEM: ::std::os::raw::c_char = -4;
-pub const JNI_EEXIST: ::std::os::raw::c_char = -5;
-pub const JNI_EINVAL: ::std::os::raw::c_char = -6;
+pub const JNI_OK: ::std::os::raw::c_int = 0;
+pub const JNI_ERR: ::std::os::raw::c_int = -1;
+pub const JNI_EDETACHED: ::std::os::raw::c_int = -2;
+pub const JNI_EVERSION: ::std::os::raw::c_int = -3;
+pub const JNI_ENOMEM: ::std::os::raw::c_int = -4;
+pub const JNI_EEXIST: ::std::os::raw::c_int = -5;
+pub const JNI_EINVAL: ::std::os::raw::c_int = -6;
 pub const JNI_COMMIT: ::std::os::raw::c_uchar = 1;
 pub const JNI_ABORT: ::std::os::raw::c_uchar = 2;
 pub const JNI_VERSION_1_1: ::std::os::raw::c_uint = 65537;
 pub const JNI_VERSION_1_2: ::std::os::raw::c_uint = 65538;
 pub const JNI_VERSION_1_4: ::std::os::raw::c_uint = 65540;
 pub const JNI_VERSION_1_6: ::std::os::raw::c_uint = 65542;
 pub const JNI_VERSION_1_8: ::std::os::raw::c_uint = 65544;
-pub type jint = ::std::os::raw::c_int;
-pub type jlong = ::std::os::raw::c_long;
+pub type jint = i32;
+pub type jlong = i64;
 pub type jbyte = ::std::os::raw::c_char;
```

Comment:
While we're at it, the other `j<type>` types should probably get the same treatment.